src/binder/clrprivbindercoreclr.cpp

```diff
@@ -157,10 +149,12 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
                 if (SUCCEEDED(hr))
                 {
+                    // Is identity same as TPA assembly
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        // Fail bind if requested assembly is at different location than TPA assembly
+                        if(pPEImage != pCoreCLRFoundAssembly->GetPEImage())
```

Comment:
What happens today if we have user assembly A1 1.0.0.0 loaded from C:\foo into a load context (Default or custom) and later get a request to load A1 1.0.0.0 from c:\foo2? Iwoudl expect that we will find a AssemblySpec match against what is already loaded and return that from the binder - is that not true? If so, why wont we follow the same pattern here?